### PR TITLE
cdc: fix old value config glitch when changefeeds with different settings connect to one region

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve?

Problem Summary:
When two TiCDC changefeeds connect to the same region, but one changefeed requires old value but the other doesn't, the one that doesn't need old value still receives old value.

### What is changed and how it works?

What's Changed: Now we record the old value setting in each `Downstream` and filters the old value appropriately before sinking the events. 

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 
- Fix old value config glitch when changefeeds with different settings connect to one region

### CODE REVIEW

File: components/cdc/tests/integrations/test_cdc.rs [hunk 1]
```diff
@@ -890,6 +890,96 @@ fn test_old_value_basic() {
...
+    m2.value = vec![b'3'; 5120];
+    let ts3 = block_on(suite.cluster.pd_client.get_tso()).unwrap();
+    suite.must_kv_prewrite(1, vec![m2], k1.clone(), ts3);
+    let ts4 = block_on(suite.cluster.pd_client.get_tso()).unwrap();
+    suite.must_kv_commit(1, vec![k1], ts3, ts4);
+    println!("test_old_value_multi_changefeeds: 2");
+    let mut event_count = 0;
+    loop {
+        let events = receive_event_1(false).events.to_vec();
+        println!("test_old_value_multi_changefeeds: 3");
```
Comment (5kbpers) [inline components/cdc/tests/integrations/test_cdc.rs:931]: Please remove these `println`

**Review by amyangfei — COMMENTED** [review]
LGTM

**Pull request merged**